federatedscope/core/monitors/monitor.py

```diff
@@ -376,6 +376,20 @@ def format_eval_res(self,
                                 all_res.size // 10]
                             new_results[f"{key}_top_decile"] = all_res[
                                 all_res.size * 9 // 10]
+                            # log more fairness metrics
+                            # min and max
+                            new_results[f"{key}_min"] = all_res[0]
+                            new_results[f"{key}_max"] = all_res[-1]
+                            #bottom and top 10%
+                            new_results[f"{key}_bottom10%"]=np.mean(all_res[:
+                                all_res.size  // 10])
+                            new_results[f"{key}_top10%"]=np.mean(all_res[
+                                all_res.size * 9 // 10:])
+                            #cosine similarity between the performance distribution and 1
+                            new_results[f"{key}_cos1"]=np.mean(all_res)/(np.sqrt(np.mean(all_res**2)))
+                            #entropy of performance distribution
+                            all_res_preprocessed=all_res+1e-9
+                            new_results[f"{key}_entropy"]=np.sum(-all_res_preprocessed/np.sum(all_res_preprocessed)*(np.log((all_res_preprocessed)/np.sum(all_res_preprocessed))))
```

Comment:
Please update the comments in line 318 according to the new metrics added: 
            'Results_fairness': {
                'test_correct': 3356,      'test_total': 5000,
                'test_avg_loss_std': 0.04, 'test_avg_loss_bottom_decile':
                0.52, 'test_avg_loss_top_decile': 0.64,
                'test_acc_std': 0.06,      'test_acc_bottom_decile': 0.60,
                'test_acc_top_decile': 0.75,
                'test_loss_std': 214.17,   'test_loss_bottom_decile':
                2644.64, 'test_loss_top_decile': 3241.23
                },
            }